framework/base/Model.php

```diff
@@ -239,9 +239,15 @@ public function scenarios()
      */
     public function formName()
     {
-        $reflector = new ReflectionClass($this);
+        static $formName = null;
```

Comment:
Isn't it a kind of a de-facto convention to rather use protected class properties in all these cases?

```
protected $_var;

public function getFoo()
{
    if ($this->_var === null) {
        $this->_var = ...
    }
    return $this->_var
}
```

I think it's effectively doing the same. But I'd vote to use one pattern only. This pattern seems to be more established in the current code base.